src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -274,7 +274,7 @@ public final void beginTree(DetailAST rootAST) {
 
     @Override
     public final void finishTree(DetailAST rootAST) {
-        TREE_CACHE.get().clear();
+        // No code by default
```

Comment:
We clear the cache in the begin tree and the finish tree. I see no reason we need both and it is more important we clear it on the begin. So I removed this.